test/test_jit.py

```diff
@@ -5952,84 +5952,86 @@ def func(a, b):
+            else:
+                checkMath(func_name, num_args, is_float, **args)
 
-        def test_floor(x):
-            # type: (float) -> int
-            return math.floor(x)
-
-        def test_ceil(x):
-            # type: (float) -> int
-            return math.ceil(x)
-
-        def test_log_int(x):
-            # type: (int) -> float
-            return math.log(x)
-
-        def test_log_float(x):
-            # type: (float) -> float
-            return math.log(x)
-
-        def test_log_base_float(x, y):
-            # type: (float, float) -> float
-            return math.log(x, y)
-
-        def test_log1p_int(x):
-            # type: (int) -> float
-            return math.log1p(x)
-
-        def test_log1p_float(x):
-            # type: (float) -> float
-            return math.log1p(x)
-
-        def test_log10_int(x):
-            # type: (int) -> float
-            return math.log10(x)
-
-        def test_log10_float(x):
-            # type: (float) -> float
-            return math.log10(x)
+        inf = float("inf")
+        NaN = float("nan")
+        mx_int = 2147483647
+        mn_int = -2147483647 - 1
+        float_vals = [inf, NaN, 0.0, 1.0, 2.2, -1.0, -0.0, -2.2, -inf, 1, 0, 2]
+        int_vals = list(range(-5, 5, 1)) + [mx_int + 5, mx_int * 2, mn_int - 5, mn_int * 2]
 
-        def test_exp_int(x):
-            # type: (int) -> float
-            return math.exp(x)
+        def checkMath(func_name, num_args, is_float=True, ret_type="float", debug=False, vals=None, args_type=None):
+            funcs_template = dedent('''
+            def func(a, b):
+                # type: {args_type} -> {ret_type}
+                return math.{func}({args})
+            ''')
+            if num_args == 1:
+                args = "a"
+            elif num_args == 2:
+                args = "a, b"
+            else:
+                raise RuntimeError("Test doesn't support more than 2 arguments")
+            if args_type is None:
+                args_type = "(float, float)" if is_float else "(int, int)"
+            funcs_str = funcs_template.format(func=func_name, args=args, args_type=args_type, ret_type=ret_type)
+            scope = {}
+            execWrapper(funcs_str, globals(), scope)
+            cu = torch.jit.CompilationUnit(funcs_str)
+            fs = cu.func
+            f = scope['func']
 
-        def test_exp_float(x):
-            # type: (float) -> float
-            return math.exp(x)
 
-        def test_sqrt_int(x):
-            # type: (int) -> float
-            return math.sqrt(x)
+            if vals is None:
+                vals = float_vals if is_float else int_vals
+                vals = [(i, j) for i in vals for j in vals]
 
-        def test_sqrt_float(x):
-            # type: (float) -> float
-            return math.sqrt(x)
+            for a, b in vals:
+                resf = None
+                resfs = None
+                try:
```

Comment:
This flow is hard to follow (and the names could be more descriptive), it seems like it could mostly be replaced with `self.assertEqual`

Reply:
There's a comment about the conditions that `self.assertEqual` doesn't support - is it possible to make it support that? https://github.com/pytorch/pytorch/pull/21125/files#diff-78aab5b4217aeaed920efab0c5f5f5c8R6009

I suppose it might be able to replace the last if statement or so...

Reply:
Oh yeah I missed that, the `NaN` thing we shouldn't change so I guess it's not all that useful,  what do you mean by `2. We want to return `nan` for math.sqrt(-5)`, shouldn't it be an exception?